LabApi/Loader/PluginLoader.cs

```diff
@@ -141,24 +143,24 @@ public static void LoadPlugins(IEnumerable<FileInfo> files)
                     // Otherwise, we load the assembly without any debug information.
                     : Assembly.Load(File.ReadAllBytes(file.FullName));
 
+                plugins.Add((file, pluginAssembly));
+            }
+            catch (Exception e)
+            {
+                Logger.Error($"{LoggerPrefix} Couldn't load the plugin inside '{file.FullName}'");
+                Logger.Error(e);
+            }
+        }
+
+        foreach ((FileInfo file, Assembly assembly) in plugins)
+        {
+            try
+            {
                 // If the assembly has missing dependencies, we skip it.
-                if (AssemblyUtils.HasMissingDependencies(pluginAssembly, file.FullName, out Type[]? types))
+                if (AssemblyUtils.HasMissingDependencies(assembly, file.FullName, out Type[]? types))
                     continue;
 
-                foreach (Type type in types)
-                {
-                    // We check if the type is derived from Plugin.
-                    if (!type.IsSubclassOf(typeof(Plugin)))
-                        continue;
-
-                    // We create an instance of the type and check if it was successfully created.
-                    if (Activator.CreateInstance(type) is not Plugin plugin)
-                        continue;
-
-                    // In that case, we add the plugin to the plugins list and log that it has been loaded.
-                    Plugins.Add(plugin, pluginAssembly);
-                    Logger.Info($"{LoggerPrefix} Successfully loaded {plugin.Name}");
-                }
+                InstantiatePlugins(types, assembly);
```

Comment:
```suggestion
                InstantiatePlugin(types, assembly);
```
This instantiates a single plugin, therefore it should refer to "Plugin" instead of multiple.

Reply:
The method may register multiple plugins if an assembly has more than one plugin defined.

Reply:
Yeah that's a point i see now
Its fine then!